src/UI/templates/default/Icon/icon.less

```diff
@@ -52,6 +52,18 @@
 			height: @icon-size-large;
 		}
 	}
+	&.responsive {
+		width: 100%;
+		line-height: @icon-size-medium;
+		.abbreviation {
+			font-size: ceil(@icon-size-medium / 2.5);
+			width: @icon-size-medium;
+		}
+		img {
+			width: 100%;
+			height: auto;
```

Comment:
Did you check if that also works with the standard version of the icon?

Reply:
The standard version does not use img tags. So these rules do not apply. Maybe we currently do not allow standard icons in the repo cards. Not sure how we get the standard version to auto-scale.

Reply:
i.O.., I you want me, I can give it a try to make this workable with the standard icon. Just give me a notification if so. Note that there is no way, this will be merged without being able to handle the standard icon.

Reply:
Yes, please try. Quite a disappointing statement though. I focused on custom icons since I thought this would be accepted, since icons are uploadable (even if not the outlined version yet).

Reply:
Hi @alex, the following points might help to ease your disappointment about my decision:

1. You need to accept both anyways. In my comment above I wrote.

> Hi @alex40724 the JF accepted the interface which requires an Icon which is either a standard or custom. There might be cases, where you will need a custom Icon.

You or @xus proposed in the PR for the interface to support "Icons" for this, it was not me.  This was fine for me and the JF. This was accepted. If you are proposing to accept Icons, you have to be able to handle them. How else, would you explain somebody, that he may stuff a standard icon in there through the interface, but this completely breaks the css? We could talk about changing the interface, if it was obvious, that clearly only the custom icons make sense here. However this is not the case. Your work on the custom Icon is not lost. You would have to handle them as well, since you accept them on the interface. 

2. You CSS currently completely breaks if no Icon is given. This is possible, again through your own proposal:

Usage
Repository Object Cards MAY contain a UI Icon displaying the object type.
Repository Object Cards MAY contain a UI ProgressMeter displaying the learning progress of the user.
Repository Object Cards MAY contain a UI Icon displaying a certificate icon if the user finished the task.
Repository Object Cards MAY contain a UI ProgressMeter OR UI Icon certificate, NOT both.

So you need to address the issue of not having a height set in the il-card-repository-head if not having an custom icon anyways.

3. This is fixable, by:
a) Giving this section a fixed height. However, we would need to make some adaptions for various possible screen AND card sizes. However, this would be solid and also make the rendering of the dropdown a little easier and better (currently it looks funky on various screen/card sizes).

b) Add a completely transparent 32-32 svg which we tell the icon to use as placolder for standard icons. This is no breaking change so no JF decision needed. However, this would not yet solve the "no-icon-at-all" case. For this you would need to add some css fixing the height. If we decide to add the placeholder svg to the standard set, you could use that.

As you see, this is an issue you need to fix regardless of the support of the standard icon question. @alex40724 which option do you prefer? @klees which option do you prefer.

Reply:
@Amstutz IMO 1. is the way to go.

Reply:
My problem is not that I have to fix things, but a) when, b) for what use cases and c) if they are fixable at all. I was not able to work out a proper soloution with a fixed height approach. Did you try this for all deck variants, did it look less funky? It is not so easy if the icons should be equally distributed in the header section. You either have lots of space and tiny icons or bigger icons.

Of course it would be a lot easier if I put all the icons, e.g. right aligned and give them fixed sizes. But this was not the idea of the initial design proposal, right?

I thought it may be acceptable to use the custom icons for now. We do not have another use case for this and could discuss how to handle standard icons later. Also the case that no icon at all is passed is not one that is needed by any feature yet.

I have no JF before beta anymore so I cannt change the rules (and e.g. only accept custom icons for now, or ask the consumer to include at least one element for the header). So I am lost, it just stalls the whole feature.

I very much appreciate if you could try to work out a CSS solution that is acceptable for you.